### PR TITLE
feat: Backlog/unreal/project context/create level publisher

### PR DESCRIPTION
**Pull request by henriknorin-ftrack**
Resolves 

## Changes

- Unreal level publisher
- Unreal batch publish of level snapshot dependencies to root project context.

## Test

_Note: needs to be tested with corresponding branch on pipeline, qt, definition and unreal repos._

Publish a level with dependencies in Unreal (PC), it should publish the level and its dependencies as snapshot, with content folder structure, beneath root context.

### CODE REVIEW

File: source/ftrack_connect_pipeline_qt/ui/utility/widget/entity_info.py [hunk 1]
```diff
@@ -77,7 +77,11 @@ def post_build(self):
...
     def _on_path_ready(self, parents):
         '''Set current path to *names*.'''
-        self._name_field.setText('{} '.format(parents[-1]['name']))
-        self._path_field.setText(
-            os.sep.join([p['name'] for p in parents[:-1]])
-        )
+        self.set_name_field(parents[-1]['name'])
+        self.set_path_field(os.sep.join([p['name'] for p in parents[:-1]]))
+
+    def set_name_field(self, name):
```
Comment (henriknorin-ftrack) [inline source/ftrack_connect_pipeline_qt/ui/utility/widget/entity_info.py:83]: Add docstrings

**Review by henriknorin-ftrack — COMMENTED** [review]
Needs changes

File: source/ftrack_connect_pipeline_qt/client/publish/batch.py [hunk 2]
```diff
@@ -0,0 +1,401 @@
...
+        self.run_plugin(plugin_data, method, self.engine_type)
+
+    def _build_definition_selector(self):
+        raise NotImplementedError()
+
+    def _build_batch_publisher_widget(self):
+        raise NotImplementedError()
+
+    def _get_list_widget_class(self):
+        raise NotImplementedError()
```
Comment (lluisFtrack) [inline source/ftrack_connect_pipeline_qt/client/publish/batch.py:249]: Are those lines needed?

File: source/ftrack_connect_pipeline_qt/client/publish/batch.py [hunk 3]
```diff
@@ -0,0 +1,401 @@
...
+
+                item_widget.finalizer_user_data = None
+
+                engine_type = definition['_config']['engine_type']
+
+                # Run the definition, status feedback will come in async but pushed to main thread by factory
+                self.run_definition(definition, engine_type)
+
+                # Did it go well?
+                print(
```
Comment (lluisFtrack) [inline source/ftrack_connect_pipeline_qt/client/publish/batch.py:319]: Remove this

**Review by lluisFtrack — CHANGES_REQUESTED** [review]

File: source/ftrack_connect_pipeline_qt/plugin/widget/base_collector_widget.py [hunk 4]
```diff
@@ -49,7 +52,7 @@ def build(self):
         self.layout().addWidget(self._summary_widget)
 
         self.list_widget = QtWidgets.QListWidget()
-        self.list_widget.setAlternatingRowColors(True)
+        # self.list_widget.setAlternatingRowColors(True)
```
[the remark this answers is not in the document]
Comment (henriknorin-ftrack) [reply source/ftrack_connect_pipeline_qt/plugin/widget/base_collector_widget.py:55]: Sidenote: had to stop using alternating row colors as I could not get it to work with styling.

File: source/ftrack_connect_pipeline_qt/client/publish/batch.py (hunk 2 above)
Comment (henriknorin-ftrack) [reply source/ftrack_connect_pipeline_qt/client/publish/batch.py:249]: Not the last one, removing

Comment (henriknorin-ftrack) [reply source/ftrack_connect_pipeline_qt/client/publish/batch.py:249]: Added docstrings as well

**Review by lluisFtrack — APPROVED** [review]
LGTM

**Pull request merged**